### PR TITLE
Refactor game team

### PR DESCRIPTION
**Pull request by Aphilosopher30**

### CODE REVIEW

File: lib/game_team.rb [hunk 1]
```diff
@@ -2,13 +2,18 @@ class GameTeam
...
     @team_id = raw_data[:team_id].strip
     @result = raw_data[:result]
     @head_coach = raw_data[:head_coach]
     @goals = raw_data[:goals].to_i
     @shots = raw_data[:shots].to_i
     @tackles = raw_data[:tackles].to_i
-   end
+  end
+
+  def is_game_pair?(game_team)
```
Comment (ghost) [inline lib/game_team.rb:15]: Big fan of these methods that help out the managers

File: lib/game_teams_manager.rb [hunk 2]
```diff
@@ -18,6 +18,16 @@ def generate_list(data_path)
...
 
+  def most_tackles(season)
+    tackle_hash = get_team_tackle_hash(season)
+    tackle_hash.key(tackle_hash.values.max)
+  end
+
+  def fewest_tackles(season)
+    tackle_hash = get_team_tackle_hash(season)
+    tackle_hash.key(tackle_hash.values.min)
+  end
```
Comment (ghost) [inline lib/game_teams_manager.rb:29]: Nice and clean

File: lib/game_teams_manager.rb [hunk 3]
```diff
@@ -41,75 +61,61 @@ def score_and_shots_by_team(season_games_ids)
...
+      rivals[rival_game.team_id][0] += 1 if rival_game.result == "WIN"
     end
-    rivals.key(rivals.values.min)
+    rivals
+  end
+
+  def favorite_opponent(team_id)
+    rival_hash = get_hash_of_rival_teams(team_id)
+    rival_hash_ratio = create_ratio_hash(rival_hash)
+    rival_hash_ratio.key(rival_hash_ratio.values.min)
```
Comment (ghost) [inline lib/game_teams_manager.rb:112]: Yessss thank you

File: lib/stat_tracker.rb [hunk 4]
```diff
@@ -114,29 +114,25 @@ def lowest_scoring_visitor
   ###### Season Stats #######
   def most_tackles(season)
     season_games = @games.get_season_games(season)
-    tackle_hash = @game_teams.get_team_tackle_hash(season_games)
-    team_id = tackle_hash.key(tackle_hash.values.max)
+    team_id = @game_teams.most_tackles(season_games)
```
Comment (ghost) [inline lib/stat_tracker.rb:117]: very nice

**Review by ghost — APPROVED** [review]

File: lib/game_team.rb (hunk 1 above)
Comment (denverdevelopments) [reply lib/game_team.rb:15]: Good job delegating to game_team

File: lib/game_teams_manager.rb (hunk 3 above)
Comment (denverdevelopments) [reply lib/game_teams_manager.rb:112]: Very nice consolidation

**Review by denverdevelopments — APPROVED** [review]

**Pull request merged**